subt/tools/analyze_detectors.py

```diff
@@ -0,0 +1,243 @@
+"""evaluate detectors"""
+import os
+import numpy as np
+import cv2
+from matplotlib import pyplot as plt
+
+try:
+    import torch
+    import subt.artf_model
+    from subt.artf_detector import Detector
+except ImportError:
+    print('\nWarning: missing torch!\n')
+
+from osgar.logger import LogReader, lookup_stream_id
+from osgar.lib.serialize import deserialize
+from subt.tf_detector import CvDetector
+from subt.artf_node import check_results
+
+g_streams = ["logimage.image", "logimage_rear.image"]
+g_artf_names = ["survivor","backpack", "phone", "helmet", "rope", "fire_extinguisher", "drill", "vent", "cube"]
+
+NAMES_AND_SCORES = {'backpack': 0.1,
+                    'survivor': 0.1,
+                    'phone': 0.1,
+                    'rope': 0.1,
+                    'helmet': 0.1,
+                    'fire_extinguisher': 0.1,
+                    'drill': 0.1,
+                    'vent': 0.1,
+                    'cube': 0.1,
+                    'robot': 1.0,
+                    'breadcrumb': 1.0,
+                    'nothing': 1.0
+                    }
+MIN_SCORES = np.array(list(NAMES_AND_SCORES.values()))
+
+
+def take_third(item):
+    return item[2]
+
+
+def proces_torch_result(result):
+    result.sort(key=take_third, reverse=True)
+    score_torch = result[1][2]
+    points = [[x, y] for x, y, c in result][:2]
+
+    return score_torch, points
+
+
+def manual_sorting(data):
+    ii = 0
+    while True:
+        if ii < 0:
+            ii = 0
+        if ii >= len(data):
+            ii = len(data) - 1
+        artf_name, score_t, score_cv, points, bbox, im_path, detection_type = data[ii]
+        print(bbox)
+        x, y, xw, yh = bbox
+        im_path = im_path[:-1]  # remove line end
```

Comment:
`im_path = im_path.rstrip('\r\n')` looks safer to me with respect to various operating systems we use.

Reply:
yes, it looks better.